tracer/src/Datadog.Trace/Debugger/Symbols/SymbolExtractor.cs

```diff
@@ -0,0 +1,718 @@
+                        {
+                            classStartLine = innerScope.StartLine;
+                        }
+
+                        if (classEndLine < innerScope.EndLine)
+                        {
+                            classEndLine = innerScope.EndLine;
+                        }
+
+                        classSourceFile ??= innerScope.SourceFile;
+                    }
+                }
+            }
+
+            if (classStartLine == int.MaxValue)
+            {
+                classStartLine = UnknownMethodStartLine;
+            }
+
+            if (classEndLine == 0)
+            {
+                classEndLine = UnknownMethodEndLine;
+            }
+
+            return new SourcePdbInfo { StartLine = classStartLine, EndLine = classEndLine, Path = classSourceFile };
+        }
+
+        private void PopulateNestedNotCompileGeneratedClassScope(ImmutableArray<TypeDefinitionHandle> nestedTypes, Model.Scope[] nestedClassScopes, ref int index)
+        {
+            for (var i = 0; i < nestedTypes.Length; i++)
+            {
+                var typeHandle = nestedTypes[i];
+                if (typeHandle.IsNil)
+                {
+                    continue;
+                }
+
+                if (DatadogMetadataReader.IsCompilerGeneratedAttributeDefinedOnType(typeHandle.RowId))
+                {
+                    continue;
+                }
+
+                if (!TryGetClassSymbols(typeHandle, out var nestedClassScope))
+                {
+                    continue;
+                }
+
+                nestedClassScopes[index] = nestedClassScope.Value;
+                index++;
+            }
+        }
+
+        private LanguageSpecifics GetClassLanguageSpecifics(TypeDefinition type)
+        {
+            var interfaceNames = GetClassInterfaceNames(type);
+
+            var baseClassNames = GetClassBaseClassNames(type);
+
+            var accessModifiers = (type.Attributes & TypeAttributes.VisibilityMask);
+            var classLanguageSpecifics = new LanguageSpecifics
+            {
+                AccessModifiers = accessModifiers > 0 ? new[] { accessModifiers.ToString() } : null,
+                Interfaces = interfaceNames,
+                SuperClasses = baseClassNames
+            };
+            return classLanguageSpecifics;
+        }
+
+        private Symbol[] GetFieldSymbols(FieldDefinitionHandleCollection fieldsCollection, TypeDefinition parentType)
+        {
+            int index = 0;
+            Model.Symbol[] fieldSymbols = new Model.Symbol[fieldsCollection.Count];
+            foreach (var fieldDefHandle in fieldsCollection)
+            {
+                var fieldDef = MetadataReader.GetFieldDefinition(fieldDefHandle);
+                var fieldTypeName = fieldDef.DecodeSignature(new TypeProvider(false), 0);
+                var fieldName = fieldDef.Name.IsNil ? null : MetadataReader.GetString(fieldDef.Name);
```

Comment:
Isn't it sabotages the all point of uploading this particular symbol if the field name is unknown (`null`)?
I guess we should skip it in this case (as we do for methods for example). WDYT?

Reply:
I'm not sure I understand but because the file is outdated already, the issue might be fixed.

Reply:
That's the sentence: `var fieldName = fieldDef.Name.IsNil ? null : MetadataReader.GetString(fieldDef.Name);`

Reply:
So it's fixed